tests/system/Database/Live/UpdateTest.php

```diff
@@ -51,8 +51,15 @@ public function testUpdateSetsAllWithoutWhereAndLimit()
                 ->get()
                 ->getResult();
 
-            $this->assertSame('Bobby', $result[0]->name);
-            $this->assertSame('Ahmadinejad', $result[1]->name);
+            // this is really a bad test - indexes and other things can affect sort order
+            if ($this->db->DBDriver === 'SQLSRV') {
+                $this->assertSame('Derek Jones', $result[0]->name);
+                $this->assertSame('Bobby', $result[1]->name);
+            } else {
+                $this->assertSame('Bobby', $result[0]->name);
+                $this->assertSame('Ahmadinejad', $result[1]->name);
+            }
+
```

Comment:
If order is not consistent but we aren't testing for order then this should be refactored to something like:
```suggestion
            $this->assertEqualsCanonacalizing(['Bobby', 'Derek Jones'], array_column($result, 'name');
```
I'm on mobile so can't finish that but hopefully you get the idea.

Reply:
@MGatner 
A canonical test will fail here as well. We start with four names in the database.. none of which are Bobby. We the run a query:
UPDATE user SET name = 'Bobby' LIMIT 1
Prior to adding the indexes on the user table the database would select rows for update in a somewhat relyable fashion. It would order based on the Primary key. But after adding indexes at least SQLSRV updates on a different sort order. Depending on which name gets overwritten with Bobby you end up with differet combinations of names.

About the only relyable assertions that can be made here is that there should be one and only one user with the name Bobby. 

I could change to use jobs table which only has primary key, leave as is ..as i have worked out the order in which they get updated, or simplify the assertion that only one Bobby does exist. 

This sort of query is rather odd and would rarely be used. Depending on how indexing is setup it might be more useful in some circumstances.

Reply:
https://www.brentozar.com/archive/2019/05/why-order-isnt-guaranteed-without-an-order-by/

Reply:
It seems there is no way to improve the test.
The test is unrealistic, but it tests the case of `testUpdateSetsAllWithoutWhereAndLimit`.

Reply:
Got it! I understand now, thanks for the explanation.